experiment/measurer/measure_manager.py

```diff
@@ -580,20 +522,53 @@ def process_crashes(self, cycle):
                              crash_stacktrace=crash.crash_stacktrace))
         return crashes
 
-    def update_measured_files(self):
-        """Updates the measured-files.txt file for this trial with
-        files measured in this snapshot."""
-        current_files = set(os.listdir(self.corpus_dir))
-        already_measured = self.get_measured_files()
-        filesystem.write(self.measured_files_path,
-                         '\n'.join(current_files.union(already_measured)))
-
-    def get_measured_files(self):
-        """Returns a the set of files that have been measured for this
-        snapshot's trials."""
-        if not os.path.exists(self.measured_files_path):
-            return set()
-        return set(filesystem.read(self.measured_files_path).splitlines())
+    def save_detailed_coverage_files_state(self, cycle):
+        """Saves the measured-detailed-coverage-files StateFile for this cycle
+        with the detailed coverage files measured in this cycle and previous
+        ones."""
+        previous_files = self.get_prev_measured_files(cycle,
+                                                      detailed_coverage=True)
+        current_segment_files = set(os.listdir(self.segment_dir))
+        current_function_files = set(os.listdir(self.function_dir))
+        all_files = previous_files.union(current_function_files,
+                                         current_segment_files)
+        measured_files_state = self.get_measured_files_state(cycle)
+        measured_files_state.set_current(list(all_files))
+        return all_files
```

Comment:
So, you are storing the segment and function file *names* in the state file? Why?